### PR TITLE
Improved matchers to support array values. Squashed some bugs. Documentation improvements.

### PR DESCRIPTION
**Pull request by nbrink91**

### CODE REVIEW

File: example/tests/Provider/PactVerifyTest.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+use Symfony\Component\Process\Process;
+
+class PactVerifyTest extends TestCase
+{
+    /** @var Process */
+    private $process;
+
+    protected function setUp()
+    {
+        $this->process = new Process('php -S localhost:8080 -t ../../src/Provider/public/');
```
Comment (mattermack) [inline example/tests/Provider/PactVerifyTest.php]: @nbrink91, this is to work around using a bootstrap?

Comment (nbrink91) [reply example/tests/Provider/PactVerifyTest.php]: Yeah, let me clean that up a bit. I did that and didn't wrap it up. I need to add an assertion as well.

File: src/PhpPact/Consumer/TestCase/MockServerTestCase.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    private $mockServer;
+
+    protected function setUp()
+    {
+        $this->config = new MockServerEnvConfig();
+        $this->config
+            ->setPactFileWriteMode('merge')
+            ->setPactDir('D:/Development/PACT/pact-php');
+        $this->mockServer = new MockServer($this->config);
+        $this->mockServer->start();
```
Comment (mattermack) [inline src/PhpPact/Consumer/TestCase/MockServerTestCase.php]: I thought the start and stop was in the listener (i.e. reason to move away from the bootstrap)

File: src/PhpPact/Consumer/TestCase/MockServerTestCase.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    protected $config;
+    /** @var MockServer */
+    private $mockServer;
+
+    protected function setUp()
+    {
+        $this->config = new MockServerEnvConfig();
+        $this->config
+            ->setPactFileWriteMode('merge')
+            ->setPactDir('D:/Development/PACT/pact-php');
```
Comment (mattermack) [inline src/PhpPact/Consumer/TestCase/MockServerTestCase.php]: Can this be a relative path?

File: src/PhpPact/Consumer/TestCase/MockServerTestCase.php (hunk 2 above)
Comment (nbrink91) [reply src/PhpPact/Consumer/TestCase/MockServerTestCase.php]: You are right, I added this when playing around with the merge ability of the ruby mock server. I deleted the file. Thanks!

File: src/PhpPact/Consumer/TestCase/MockServerTestCase.php (hunk 3 above)
Comment (nbrink91) [reply src/PhpPact/Consumer/TestCase/MockServerTestCase.php]: Yes, you can definitely use a relative path. This file should have never been committed. It was just something I was experimenting with.

File: src/PhpPact/PactBrokerConnector.php [hunk 4]
```diff
@@ -59,6 +59,7 @@ public function publishFile($file, $version)
 
     /**
      * @deprecated This will be removed in the next major release. @see BrokerHttpService::publishJson for the replacement.
```
Comment (mattermack) [inline src/PhpPact/PactBrokerConnector.php]: Nice, @nbrink91

**Pull request merged**